File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        indexi=-1    
        for i,j in enumerate(nums):
            sumi=0
            while(j>0):
                a=j%10
                sumi+=a
                j=j//10
                
            if i == sumi:
                
                indexi=i
                break
        return indexi
        
        

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        int index = -1;
        for (int i = 0; i < nums.length; i++) {
            int digitSum = 0;
            int num = nums[i];
            while (num > 0) {
                digitSum += num % 10;
                num /= 10;
            }
            if (i == digitSum) {
                index = i;
                break;
            }
        }
        return index;
    }
        
    }
